### PR TITLE
Add new 'centerMode' feature

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

We don't currently support sliders that behave like this:

This is something users have been asking for a long time, this PR enables us to do so, through the use of a new `centerMode` prop.

Also, we now support this version as well (setting `centerMode` to `to-the-left`):

#### How to test it?

Check this [Workspace](https://sliderlayoutcentermode--storecomponents.myvtex.com/) out. You should see that the shelf looks normal on a desktop. Switch to mobile and you can see the new behavior, where there is a "preview" of the next and previous items.

Just to illustrate the case where I think this feature works best, I used the following `slider-layout` props to achieve the behavior shown in the workspace.

```json
{
  "slider-layout#demo-products": {
    "props": {
      "itemsPerPage": {
        "desktop": 5,
        "tablet": 3,
        "phone": 1
      },
      "infinite": true,
      "fullWidth": false,
      "blockClass": "shelf",
      "showNavigationArrows": "desktopOnly",
      "centerMode": {
        "desktop": "disabled",
        "mobile": "center"
      }
    }
  },
}
```

#### Screenshots or example usage:

**Phone**
Screen Shot 2020-08-18 at 18 24 28

**Tablet**
Screen Shot 2020-08-18 at 18 24 01

**Desktop**
Screen Shot 2020-08-18 at 18 22 41

#### How does this PR make you feel? [:link:](http://giphy.com/)

Math

### CODE REVIEW

File: react/components/SliderContext.tsx [hunk 1]
```diff
@@ -206,19 +207,32 @@ const SliderContextProvider: FC<SliderContextProps> = ({
...
   const transformMap = useMemo(() => {
     const currentMap: Record<number, number> = {}
 
     newSlides.forEach((_, idx) => {
       const currIdx = infinite ? idx - resolvedSlidesPerPage : idx
+      const transformValue =
+        centerMode !== 'disabled'
+          ? -((1 + 1 / (4 * resolvedSlidesPerPage)) * slideWidth * idx) +
+            (centerMode === 'center' ? (slideWidth * 3) / 8 : 0)
+          : -(slideWidth * idx)
```
Comment (klzns) [inline react/components/SliderContext.tsx:229]: Break this down, example:
```tsx
      let transformValue = -(slideWidth * idx)
      if (centerMode !== 'disabled) {
        let somethingIDontKnow = -((1 + 1 / (4 * resolvedSlidesPerPage)) * slideWidth * idx)
        let somethingElse = (centerMode === 'center' ? (slideWidth * 3) / 8 : 0)
        transformValue =  somethingIDontKnow + somethingElse
      }
```

File: react/components/SliderContext.tsx [hunk 2]
```diff
@@ -206,19 +207,49 @@ const SliderContextProvider: FC<SliderContextProps> = ({
...
+          slideWidth *
+          idx
+        )
+
+        // This is a correction factor to center the slides when centerMode
+        // is enabled and set to 'center'.
+        const transformCenterCorrection =
+          centerMode === 'center' ? (slideWidth * 3) / 8 : 0
+
+        transformValue = adjustedTransformValue + transformCenterCorrection
```
Comment (kaisermann) [inline react/components/SliderContext.tsx:245]: What do you think about refactoring it to something like:

```suggestion
transformValue = adjustedTransformValue 

if (centerMode === 'center') {
  transformValue = (slideWidth * 3) / 8
}
```

?

File: react/components/SliderContext.tsx [hunk 3]
```diff
@@ -206,19 +207,49 @@ const SliderContextProvider: FC<SliderContextProps> = ({
...
 
-  const slideWidth = useMemo(() => 100 / newSlides.length, [newSlides.length])
+  const slideWidth = useMemo(() => {
+    const baseSlideWidth = 100 / newSlides.length
+
+    let resultingSlideWidth = baseSlideWidth
+
+    if (centerMode !== 'disabled') {
+      resultingSlideWidth =
+        (resolvedSlidesPerPage / (resolvedSlidesPerPage + 1)) * baseSlideWidth
```
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/components/SliderContext.tsx:217]: I saw this one yesterday!! Laughed a lot 😂

File: react/components/SliderContext.tsx (hunk 2 above)
Comment (victorhmp) [reply react/components/SliderContext.tsx:245]: I went for the least amount of reassignments to the `transformValue` variable... but I could change to what you've suggested. Do you think it improves readability?

Comment (kaisermann) [reply react/components/SliderContext.tsx:245]: I think it separates more clearly what's happening when 👀  But it's up to you

Comment (victorhmp) [reply react/components/SliderContext.tsx]: Okay! I've just changed it :)

File: docs/README.md [hunk 4]
```diff
@@ -76,6 +76,7 @@ Now, you are able to use all blocks exported by the `slider-layout` app. Check o
 | `autoplay`  | `object` | Controls the autoplay feature behavior. For more on this, check out the `autoplay` object section below.   | `undefined` |
 | `fullWidth`            | `boolean` | Whether the slides should occupy the full page width, making the arrows appear on top of them (`true`) or not (`false`). |`true` |
 | `arrowSize`            | `number` / `object`   | Slider arrows size (height and width) in pixels. This is a responsive prop, which means you can pass to it an object with different values for each breakpoint (`desktop`, `tablet`, and `phone`).  | `25`  |
```
[the remark this answers is not in the document]
Comment (victorhmp) [reply docs/README.md:78]: Yep! I'll add it :)

**Pull request merged**